src/components/PostsList.tsx

```diff
@@ -1,56 +1,58 @@
 import classNames from 'classnames';
 import React from 'react';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { set as setSelectedPost } from '../features/selectedPost/selectedPost';
 import { Post } from '../types/Post';
 
-type Props = {
-  posts: Post[],
-  selectedPostId?: number,
-  onPostSelected: (post: Post | null) => void,
-};
+export const PostsList: React.FC = () => {
+  const { posts } = useAppSelector(state => state.posts);
+  const dispatch = useAppDispatch();
+  const { selectedPost } = useAppSelector(state => state.selectedPost);
 
-export const PostsList: React.FC<Props> = ({
-  posts,
-  selectedPostId = 0,
-  onPostSelected,
-}) => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+  const onPostSelected = (post: Post | null) => {
+    dispatch(setSelectedPost(post));
```

Comment:
Change the `onPostSelected` function to toggle the selected post when the same post is clicked again.